### PR TITLE
citrus api on windows hosts

### LINKED ISSUE #1200: Release 4.3.0 not loading citrus-api on Windows machines
**Citrus Version**

[Release 4.3.0](https://github.com/citrusframework/citrus/releases/tag/v4.3.0).

**Expected behavior**

[Citrus-Simulator](https://github.com/citrusframework/citrus-simulator) (or tests, but I didn't test that) starts on Windows hosts.

**Actual behavior**

It does not, because https://github.com/citrusframework/citrus/commit/bd7ebff8c34909b0d17c399e7c8b9349c2562d0a introduces a bug: [`Path#of`](https://github.com/citrusframework/citrus/commit/bd7ebff8c34909b0d17c399e7c8b9349c2562d0a#diff-fd5f51db9f96ccee8379c970c767795e56722f46ec971a1d474359886eaa04fbR120) throws on Windows, because the `:` is an illegal character:

```shell
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/Users/[...]/4.3.0/citrus-api-4.3.0.jar
```

`new File(ROOT.toString().substring("file:".length())).toPath()` should be used instead.

### PR DESCRIPTION
**Pull request by bbortt**
I've tested the fix within the project where the problem first occurred.

relevant resources:

* https://bugs.openjdk.org/browse/JDK-8314511

closes #1200.

### CODE REVIEW

**Review by christophd — APPROVED** [review]

File: core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java [hunk 1]
```diff
@@ -116,10 +118,10 @@ public class ResourcePathTypeResolver implements TypeResolver {
     static {
         if (ROOT_IS_JAR) {
             try {
-                rootFs = FileSystems.newFileSystem(
-                    Path.of(ROOT.toString().substring("file:".length())));
+                rootFs = newFileSystem(
+                    new File(ROOT.toString().substring("file:".length())).toPath());
```
Comment (tschlat) [inline core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java:122]: Just curious. Why this? While is seems to serve the same purpose as the previous line, it mixes nio with io which seems less clean to me.

File: core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java [hunk 2]
```diff
@@ -154,8 +154,8 @@ public <T> T resolve(String resourcePath, String property, Object... initargs) {
         String cacheKey = toCacheKey(resourcePath, property, "NO_KEY_PROPERTY");
 
         Map<String, String> map = typeCache.computeIfAbsent(
-                cacheKey,
-                key -> Collections.singletonMap(key, resolveProperty(resourcePath, property))
+            cacheKey,
+            key -> Collections.singletonMap(key, resolveProperty(resourcePath, property))
```
Comment (tschlat) [inline core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java:158]: Would be OK to use a static import here as well.

File: core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java [hunk 3]
```diff
@@ -177,59 +177,56 @@ public <T> Map<String, T> resolveAll(String path, String property, String keyPro
...
 
-                        if (property.equals(TYPE_PROPERTY_WILDCARD)) {
-                            Properties properties = readAsProperties(fullPath + "/" + fileName);
-                            for (Map.Entry<Object, Object> prop : properties.entrySet()) {
-                                String type = resolveProperty(fullPath + "/" + fileName, prop.getKey().toString());
-                                typeLookup.put(fileName + "." + prop.getKey().toString(), type);
-                            }
+                    if (property.equals(TYPE_PROPERTY_WILDCARD)) {
+                        Properties properties = readAsProperties(fullPath + "/" + fileName);
+                        for (Map.Entry<Object, Object> prop : properties.entrySet()) {
```
Comment (tschlat) [inline core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java:200]: import Entry

File: core/citrus-spring/src/main/java/org/citrusframework/spi/SpringResourceWrapper.java [hunk 4]
```diff
@@ -68,11 +64,7 @@ public long contentLength() throws IOException {
         try {
             return delegate.getFile().length();
         } catch (Exception e) {
-            try {
-                return Files.size(Path.of(delegate.getLocation()));
-            } catch (NoSuchFileException ex) {
-                throw new FileNotFoundException(ex.getMessage());
-            }
+            return Files.size(new File(delegate.getLocation()).toPath());
```
Comment (tschlat) [inline core/citrus-spring/src/main/java/org/citrusframework/spi/SpringResourceWrapper.java:67]: Also here. What is the advantage of using the one over the other?

**Pull request merged**